src/main/java/com/minecolonies/coremod/commands/colonycommands/RefreshColonyCommand.java

```diff
@@ -45,15 +53,61 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
         return super.getCommandUsage(sender) + "<ColonyId|OwnerName>";
     }
 
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final List<ActionArgument> actionArgumentList,
+            @NotNull final Map<String, Object> argumentValueByActionArgumentNameMap) throws CommandException
+    {
+        Colony colony = null;
+        final Object colonyObject = argumentValueByActionArgumentNameMap.get("colony");
+        if (null != colonyObject)
+        {
+            colony = (Colony) colonyObject;
+        }
+
+        if (null == colony)
+        {
+            EntityPlayer player = null;
+            final Object playerObject = argumentValueByActionArgumentNameMap.get("player");
+            if (null != playerObject)
+            {
+                player = (EntityPlayer) playerObject;
+                if (player != null)
+                {
+                    IColony iColony = ColonyManager.getIColonyByOwner(server.getEntityWorld(), player);
+                    if (null == iColony)
+                    {
+                        if (sender instanceof EntityPlayer)
+                        {
+                            final Entity senderEntity = sender.getCommandSenderEntity();
+                            if (senderEntity != null)
+                            {
+                                final UUID mayorID = senderEntity.getUniqueID();
+                                if (iColony == null)
+                                {
+                                    iColony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), mayorID);
+                                }
+                            }
+                        }
+                    }
+
+                    if (null != iColony)
+                    {
+                        colony = ColonyManager.getColony(iColony.getID());
+                    }
+                }
+            }
+        }
+
+        executeShared(server, sender, colony);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "executeShared()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)